cpmpy/solvers/gurobi.py

```diff
@@ -0,0 +1,376 @@
+        then post those primitive constraints directly to the native solver
+
+        :param cpm_con CPMpy constraint, or list thereof
+        :type cpm_con (list of) Expression(s)
+        """
+        # add new user vars to the set
+        self.user_vars.update(set(get_variables(cpm_con)))
+
+        # apply transformations, then post internally
+        # XXX chose the transformations your solver needs, see cpmpy/transformations/
+
+        cpm_cons = flatten_constraint(cpm_con)
+        cpm_cons = linearize_constraint(cpm_cons)
+
+        for con in cpm_cons:
+            self._post_constraint(con)
+
+        return self
+
+    def solve(self, time_limit=None, solution_callback=None, **kwargs):
+        """
+            Call the gurobi solver
+
+            Arguments:
+            - time_limit:  maximum solve time in seconds (float, optional)
+            - kwargs:      any keyword argument, sets parameters of solver object
+
+            Arguments that correspond to solver parameters:
+            <Please document key solver arguments that the user might wish to change
+             for example: log_output=True, var_ordering=3, num_cores=8, ...>
+            <Add link to documentation of all solver parameters>
+        """
+        from gurobipy import GRB
+
+        if time_limit is not None:
+            raise NotImplementedError("TEMPLATE: TODO, implement time_limit")
+
+        # call the solver, with parameters
+        for param, val in kwargs.items():
+            self.gbi_model.setParam(param, val)
+
+        _ = self.gbi_model.optimize(callback=solution_callback)
+
+        my_status = self.gbi_model.Status
+
+        # new status, translate runtime
+        self.cpm_status = SolverStatus(self.name)
+        self.cpm_status.runtime = self.gbi_model.runtime
+
+        # translate exit status
+        if my_status == GRB.OPTIMAL and self._objective_value is None:
+            self.cpm_status.exitstatus = ExitStatus.FEASIBLE
+        elif my_status == GRB.OPTIMAL and self._objective_value is not None:
+            self.cpm_status.exitstatus = ExitStatus.OPTIMAL
+        elif my_status == GRB.INFEASIBLE:
+            self.cpm_status.exitstatus = ExitStatus.UNSATISFIABLE
+        else:  # another?
+            raise NotImplementedError(
+                f"Translation of gurobi status {my_status} to CPMpy status not implemented")  # a new status type was introduced, please report on github
+
+        # True/False depending on self.cpm_status
+        has_sol = self._solve_return(self.cpm_status)
+
+        # translate solution values (of user vars only)
+        if has_sol:
+            # fill in variable values
+            for cpm_var in self.user_vars:
+                solver_val = self.solver_var(cpm_var).X
+                if cpm_var.is_bool():
+                    cpm_var._value = bool(solver_val)
+                else:
+                    cpm_var._value = int(solver_val)
+
```

Comment:
here, within the 'if has_sol', you should get the objective_value from gbi_model and put it in self._objective_value; see ortools interface